crypto/evp/evp_lib.c

```diff
@@ -1226,4 +1226,56 @@ EVP_PKEY *EVP_PKEY_Q_keygen(OSSL_LIB_CTX *libctx, const char *propq,
     return ret;
 }
 
+EVP_PKEY *EVP_PKEY_generate_public_key(OSSL_LIB_CTX *libctx, const char *propq,
```

Comment:
EVP_PKEY_new_public_key_from_private() maybe?

Reply:
Maybe EVP_PKEY_compute_public_key(). Although that could be confused with the DH_compute_key() operation.